octree/include/pcl/octree/octree_pointcloud.h

```diff
@@ -295,7 +295,7 @@ class OctreePointCloud : public OctreeT {
    *  \param[in] point_idx_arg index of point addressing the voxel to be deleted.
    */
   void
-  deleteVoxelAtPoint(const int& point_idx_arg);
+  deleteVoxelAtPoint(const index_t& point_idx_arg);
```

Comment:
uindex_t?

Reply:
This is a reference, and all of our indices are `index_t`. Having an unsigned reference to a signed value muddies things up.

I've converted all the by-value to unsigned, but the reference ones can't be helped